src/App.jsx

```diff
@@ -14,57 +15,72 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [good, selectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+  return (
+    <div className="App">
+      <main className="section container">
+        {good ? (
+          <h1 className="title is-flex is-align-items-center">
+            {good} is selected
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => {
+                selectedGood('');
+              }}
+            />
+          </h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )}
 
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(goodOfGoods => (
+              <tr
+                data-cy="Good"
+                key={goodOfGoods}
+                className={
+                  goodOfGoods === good ? 'has-background-success-light' : ''
+                }
+              >
+                <td>
+                  {good !== goodOfGoods ? (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => {
+                        selectedGood(goodOfGoods);
+                      }}
```

Comment:
The `onClick` handler for the add button correctly updates the selected good. Ensure that the logic for selecting a good is consistent with the intended user experience.